src/FakeItEasy/Creation/DummyValueResolver.cs

```diff
@@ -95,6 +101,12 @@ private class ResolveByActivatingValueTypeStrategy : ResolveStrategy
             {
                 if (typeOfDummy.IsValueType && typeOfDummy != typeof(void))
                 {
+#if !TRY_ISBYREF_CREATION
+                    if (typeOfDummy.IsByRefLike)
+                    {
+                        return CreationResult.PermanentlyFailedToCreateDummy(typeOfDummy, "It is byref-like.");
+                    }

```

Comment:
Why not perform this check earlier, before we attempt any creation strategy?

Reply:
Oh, not addressed in my new PR either. Can do.